### DISCUSSION TITLE
Map Interface to concrete Type failed in v11.0.0

### DISCUSSION THREAD

**Discussion by MM-RYO** [LuckyPennySoftware/AutoMapper#3868] [NA]
Hi,
I want to map Entity1 to Entity1Map and TagData to TagData2. 
the TagData2 is implemented IData2 you can see the mapping configuration below.
It works in v10.2.2 not in v11.0.0. I also tested it in 11.0.1-alpha.0.9. it doesn't work.

The Generic type mapping also changed in v11.0.0
I can do the below mapping for Generic type in v10
```csharp
CreateMap(typeof(type1<>), typeof(type2<>));
```
but now I have to specify the type 
CreateMap(typeof(type1<int>), typeof(type1<int>));

### Source/destination types
Try to map Entity1 to Entity1Map
```csharp
  var testService = new TestService();
  var entities = testService.GetEntity1s();
  var mapEntities = _mapper.Map<IEnumerable<Entity1Map>>(entities);
```

### Mapping configuration

```csharp
CreateMap<Entity1, Entity1Map>();
CreateMap<TagData, TagData2>();
```

### Expected behavior

Should be able to map to target type

### Actual behavior

Error mapping types.
Missing type map configuration or unsupported mapping.

Mapping types:
TagData -> IData2
Project1.TagData -> Project1.IData2

Destination Member:
Tags

### Steps to reproduce
Framework: .net6
IDE: VS2022

I have attached a demo project
[Project1.zip](https://github.com/AutoMapper/AutoMapper/files/7932794/Project1.zip)

1. Launch the project wait a moment and let it be ready
2. Click 'Fetch data' menu on the top right corner
3. Go to MapTestController and TestProfile to check codes for more info

**Comment by lbargaoanu** [comment]
That repro is too complicated. You need to make it like the example below.
Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by MM-RYO** [comment]
Hi,
I converted the demo to the below codes.
I want to know whether we need to config something for Generic type and interface.

```csharp
using AutoMapper;


var config = new MapperConfiguration(c =>
{
    c.CreateMap<Entity1, Entity1Map>();
    c.CreateMap<TagData, TagData2>();
});

config.AssertConfigurationIsValid();

var mapper = config.CreateMapper();


try
{

    var testService = new TestService();
    var entities = testService.GetEntity1s();
    var mapEntities = mapper.Map<IEnumerable<Entity1Map>>(entities);
}
catch (Exception ex)
{
    ex.ToString();
} 

public class Entity1Map
{
    public int Id { get; set; }
    public string Name { get; set; }
    public IData2 Tags { get; set; }
}

public class Entity1
{
    public int Id { get; set; }
    public string Name { get; set; }

    public TagData Tags { get; set; }
}

public class TagData
{
    public string Tag { get; set; }
}

public class TagData2 : IData2
{
    public string Tag { get; set; }
}

public interface IData2
{
}

public class TestService
{
    public IEnumerable<Entity1> GetEntity1s()
    {
        return new List<Entity1>
        {
            new Entity1
            {
                Id = 1,
                Name = "Entity1",
                Tags = new TagData
                {
                    Tag = "TestData"
                }
            }
        };
    }
}
```

**Comment by lbargaoanu** [comment]
You need

     cfg.CreateMap<TagData, TagData2>();
     cfg.CreateMap<TagData, IData2>().As<TagData2>();

But I don't see what generics have to do with this repro.

**Reply by MM-RYO** [reply]
I didn't put it in demo. It's in our project. it works fine in v10.1.1. after I upgraded to the latest version it doesn't work anymore.
we didn't change any codes.

**Reply by lbargaoanu** [reply]
That doesn't mean much. If you can make a proper repro, I'll take a look.
